test/Validation/Parser.cpp

```diff
@@ -0,0 +1,363 @@
+/// Apache License 2.0
+
+#include <Validation/Parser.hpp>
+
+namespace ostk
+{
+namespace astro
+{
+namespace validation
+{
+
+Dictionary Parser::ParseYaml(const String& aPathToData, const String& aScenarioName)
+{
+    const String scenarioPath = String::Format("{0}/scenarios/{1}.yaml", aPathToData, aScenarioName);
+
+    const File scenarioFile = File::Path(Path::Parse(scenarioPath));
+
+    return CntrObject::Load(scenarioFile, CntrObject::Format::YAML).accessDictionary();
+}
+
+Table Parser::ParseCSV(const String& aPathToData, const String& aScenarioName, const Tool& aTool)
+{
+    String csvPath;
+    if (aTool == Tool::GMAT)
+    {
+        csvPath = String::Format("{0}/gmat_astrodynamics/{1}.csv", aPathToData, aScenarioName);
+    }
+    else if (aTool == Tool::OREKIT)
+    {
+        csvPath = String::Format("{0}/orekit_astrodynamics/{1}.csv", aPathToData, aScenarioName);
+    }
+    else
+    {
+        throw ostk::core::error::runtime::Wrong("Tool not recognized.");
+    }
+
+    return Table::Load(File::Path(Path::Parse(csvPath)), Table::Format::CSV, true);
+}
+
+Pair<State, SatelliteSystem> Parser::CreateInitialStateAndSatelliteSystem(const Dictionary& aDictionary)
+{
+    const Dictionary spacecraft = aDictionary["data"]["spacecraft"].accessDictionary();
+
+    const Composite satelliteGeometry(Cuboid(
+        {0.0, 0.0, 0.0}, {Vector3d {1.0, 0.0, 0.0}, Vector3d {0.0, 1.0, 0.0}, Vector3d {0.0, 0.0, 1.0}}, {1.0, 2.0, 3.0}
+    ));
+
+    PropulsionSystem propulsionSystem = PropulsionSystem::Undefined();
+    if (spacecraft.hasValueForKey("thruster"))
+    {
+        propulsionSystem = PropulsionSystem(
+            spacecraft["thruster"]["data"]["thrust"].accessReal(),
+            spacecraft["thruster"]["data"]["specific-impulse"].accessReal()
+        );
+    }
+
+    // Make entire satellite wet mass
+    const Mass wetMass = Mass::Kilograms(spacecraft["mass"].accessReal());
+
+    const SatelliteSystem satelliteSystem = SatelliteSystem(
+        Mass::Kilograms(0.0),
+        satelliteGeometry,
+        Matrix3d::Identity(),
+        spacecraft["drag-cross-section"].accessReal(),
+        spacecraft["drag-coefficient"].accessReal(),
+        propulsionSystem
+    );
+
+    const Dictionary orbit = spacecraft["orbit"].accessDictionary();
+
+    Instant initialInstant = Instant::Undefined();
+    if (orbit["data"]["date"]["time-scale"].accessString() == "UTC")
+    {
+        initialInstant = Instant::DateTime(
+            DateTime::Parse(orbit["data"]["date"]["value"].accessString(), DateTime::Format::ISO8601), Scale::UTC
+        );
+    }
+    else
+    {
+        throw ostk::core::error::runtime::Wrong("Time scale not yet supported.");
+    }
+
+    Shared<const Frame> frame = orbit["data"]["frame"].accessString() == "GCRF" ? Frame::GCRF() : Frame::Undefined();
```

Comment:
```suggestion
    const Shared<const Frame> frame = orbit["data"]["frame"].accessString() == "GCRF" ? Frame::GCRF() : Frame::Undefined();
```
Do we want to support undefined frames? O_o Should we raise an error instead?

Reply:
I figured it would error if the frame was undefined and you tried to use it later on. But maybe I should be more explicit in throwing an error here.